.github/CONTRIBUTING.md

```diff
@@ -216,6 +216,25 @@ Instead of this you should;
 	do something different
 ````
 
+### Proc arguments
+Keep typed arguments consistent across redefinitions, based on what minimal type the proc is called with rather than the actual type of the passed argument.
+````DM
+/atom/proc/do_a_thing(mob/user)
+	do something
+
+/atom/movable/do_a_thing(mob/living/carbon/human/user)
+	do something
+````
+Instead of this you should do it like this:
+````DM
+/atom/proc/do_a_thing(mob/user)
+	do something
+
+/atom/movable/do_a_thing(mob/user)
+	optional istype() check
```

Comment:
```suggestion
	var/mob/living/carbon/human/human_user = user
```

Reply:
removing the istype there would be misleading since this would be basically assuming the person is human rather than checking it

Reply:
also the istype is only optional if you need a different type than /mob

Reply:
If the proc is changing the type, it's because it needs to be typecast. The type check is assumed, and may not be necessary.